xds/internal/client/rds.go

```diff
@@ -144,3 +146,18 @@ func getClusterFromRouteConfiguration(rc *xdspb.RouteConfiguration, target strin
 	}
 	return ""
 }
+
+// hostFromTarget calls net.SplitHostPort and returns the host.
+//
+// It returns the original string instead of error if port is missing.
+func hostFromTarget(target string) (string, error) {
+	const portMissingErrDesc = "missing port in address"
+	h, _, err := net.SplitHostPort(target)
+	if err != nil {
```

Comment:
Should we check if the returned error is of type [net.AddrError](https://golang.org/pkg/net/#AddrError)?

Reply:
Done